apps/web/app/(app)/[emailAccountId]/assistant/RuleForm.tsx

```diff
@@ -1393,3 +1470,111 @@ function VariableProTip() {
     </div>
   );
 }
+
+function DelayInputControls({
+  index,
+  delayInMinutes,
+  setValue,
+}: {
+  index: number;
+  delayInMinutes: number | null | undefined;
+  setValue: ReturnType<typeof useForm<CreateRuleBody>>["setValue"];
+}) {
+  const delayConfig = useMemo(() => {
+    // Convert minutes to display value and unit
+    const getDisplayValueAndUnit = (minutes: number | null | undefined) => {
+      if (minutes === null || minutes === undefined)
+        return { value: "", unit: "hours" };
+      // -1 means enabled but empty input
+      if (minutes === -1 || minutes <= 0) return { value: "", unit: "hours" };
+
+      if (minutes >= 1440 && minutes % 1440 === 0) {
+        // Days (1440 minutes = 1 day)
+        return { value: (minutes / 1440).toString(), unit: "days" };
+      } else if (minutes >= 60 && minutes % 60 === 0) {
+        // Hours (60 minutes = 1 hour)
+        return { value: (minutes / 60).toString(), unit: "hours" };
+      } else {
+        // Minutes
+        return { value: minutes.toString(), unit: "minutes" };
+      }
+    };
+
+    // Convert display value and unit to minutes
+    const convertToMinutes = (value: string, unit: string) => {
```

Comment:
i think you can pull these helper functions out and put them at the bottom of the file. will make the useMemo much more concise and readble.

can likely drop most of the comments here too.